src/node.cc

```diff
@@ -768,6 +759,12 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
                                       Environment::kOwnsInspector));
   env.Start(per_process::v8_is_profiling);
   env.ProcessCliArgs(args, exec_args);
+  env.GetAsyncRequest()->Install(
+      &env, static_cast<void*>(&env), [](uv_async_t* handle) {
+        Environment* env = static_cast<Environment*>(handle->data);
+        uv_stop(env->event_loop());
+      });
+  uv_unref(reinterpret_cast<uv_handle_t*>(env.GetAsyncRequest()->GetHandle()));
```

Comment:
This should probably happen during `Environment::Start()`, when the other handles are created as well.

That way this could also be re-used for Workers, rather than having diverging code paths.